docs/_conversion/maintaining.md

```diff
@@ -0,0 +1,140 @@
+
+## Differences when writing custom scripting code
+
+The following applies when you're building _new_ triggers and actions, and working with custom scripting code.  These are the idioms of the way operations are executed on the new platform.  
+
+_If you're making changes to the code in the Advanced > "Legacy Web Builder" section, the following statemetns do not apply.  That code is run in an "emulated" context and works just like the old environment._
+
+**Code us scoped to the individual trigger or action**
+
+When you're writing custom scripting code in the new platform the scope of that code is local to that trigger or action.  This makes it easier to understand when a request will be handled by custom code, or simply configuring a request for default request handling.  It also means you can't easily share code between different triggers and actions.  For that, have a look at moving to the Zapier CLI, which is geared toward teams, projects with custom code, projects that need 3rd party libraries, etc.  
+
+![](https://cdn.zappy.app/27b0a08799b047fc9db1ed204d030377.gif)
+
+>Pro Tip: A handy way to understand this is to have a look at the [schema](https://zapier.github.io/zapier-platform-schema/build/schema.html) of the integration defintion you're building.  [Here](https://zapier.github.io/zapier-platform-schema/build/schema.html#basicpollingoperationschema) you'll see the structure of the definion of a polling trigger.  Notice that the `perform` field takes a request configuration object, or it simply takes a Javascript function.  The new builder UI reflects this.  When you configure an API interaction, you use the form UI to configure a request, or you use "Code Mode" to write a function to be called instead.
+
+**No pre or post functions**
+
+In Web Builder, you could write custom code that ran before the request was made, or after the request returned, or you could take over complete control of makeing the request and returning data to Zapier.  In the new platform there is only the 'complete control' option.  Your request can implement a `perform` function that takes a bundle as input and is expected to return data when the request handling is completed.  
+
+**Promise based API**
+
+A big difference in the API of the new environment is that the request API is ansyncronous, using promises rather than syncronous requests or callbacks.  For developers new to Javascript promises this can be a small stumbling block.  We recommend spending a few minutes getting familiar with how promises work in general.  The `z.request` library works very similarly to the promise-based Fetch API, so most articles and tutorials about that apply to working with custom code in Zapier's environment.  
```

Comment:
```suggestion
A big difference in the API of the new environment is that the request API is asynchronous, using promises rather than synchronous requests or callbacks.  For developers new to Javascript promises this can be a small stumbling block.  We recommend spending a few minutes getting familiar with how promises work in general.  The `z.request` library works very similarly to the promise-based Fetch API, so most articles and tutorials about that apply to working with custom code in Zapier's environment.  
```